### PR TITLE
MRG allow building only one example dir

### PR DESCRIPTION
**Pull request by jasmainak**
currently, it's not possible to build only one example dir. It's an issue when you want to just see if one particular example works. Otherwise, you need to build all the examples -- which I'm never able to do. This is a bit of a hack. @agramfort @Eric89GXL let me know if this might be useful in any way.

Do this:

```sh
$ make html_dev-pattern target PATTERN='decoding/plot_'
```

use the latest master of `sphinx-gallery`

### CODE REVIEW

File: doc/conf_template.py [hunk 1]
```diff
@@ -0,0 +1,287 @@
...
+latex_use_modindex = True
+
+trim_doctests_flags = True
+
+# Example configuration for intersphinx: refer to the Python standard library.
+intersphinx_mapping = {'http://docs.python.org/': None}
+
+sphinxgallery_conf = {
+    'examples_dirs'   : ['../examples/{{directory}}'],
+    'gallery_dirs'    : ['auto_examples/{{directory}}'],
```
Comment (jasmainak) [inline doc/conf_template.py]: here's a template variable ...

File: doc/conf_template.py [hunk 2]
```diff
@@ -273,8 +273,8 @@
 intersphinx_mapping = {'http://docs.python.org/': None}
 
 sphinxgallery_conf = {
-    'examples_dirs'   : ['../examples', '../tutorials'],
-    'gallery_dirs'    : ['auto_examples', 'auto_tutorials'],
+    'examples_dirs'   : ['../examples/{{directory}}'],
+    'gallery_dirs'    : ['auto_examples/{{directory}}'],
```
Comment (agramfort) [inline doc/conf_template.py]: this is going to break. how do you build the 2 folders now?

Comment (jasmainak) [inline doc/conf_template.py]: I just updated it ... take a look

**Comment by agramfort** [comment]
this could work but my feeling is that such a hack / use case should be address in sphinx-gallery

can you open an issue there?

**Comment by agramfort** [comment]
also using an env variable is probably simpler than relying on tempita

**Comment by agramfort** [comment]
also maybe you can pass parameters in command line. See how the no-plot option is handled

**Comment by jasmainak** [comment]
It's not super clear in my head how it would work with an env variable. But I opened an issue in [sphinx gallery](https://github.com/sphinx-gallery/sphinx-gallery/issues/82). I agree it's probably best addressed there. I'll be traveling for the next 3 days, but I'll try to find some time to follow the developments here.

**Comment by agramfort** [comment]
for example at:

https://github.com/mne-tools/mne-python/blob/master/Makefile#L55

you set MNE_SKIP_TESTING_DATASET_TESTS env variable.

**Comment by agramfort** [comment]
closing?

**Comment by jasmainak** [comment]
you don't want it as an option in the makefile? I think @jaeilepp was also suggesting it would be nice. Not super-critical for me though.

**Comment by agramfort** [comment]
ok let's finish it then

**Comment by jasmainak** [comment]
@jaeilepp want to give it a try?

**Comment by jaeilepp** [comment]
Yeah, this could be useful. I still have no idea how to use this though...

**Comment by jaeilepp** [comment]
I think we can utilise this to run parallel builds on Circleci so that we manage to get under the two hour limit. @jasmainak you'll have to show me how to use this.

**Comment by jasmainak** [comment]
okay, let's sit down together. I'm having troubles too. Something looks broken in sphinx gallery.

**Comment by jasmainak** [comment]
@jaeilepp I think I figured why it wasn't running today. Basically, you have to change this name: https://github.com/mne-tools/mne-python/blob/master/doc/conf.py#L275 to `sphinx_gallery_conf`. Does that fix it?

**Comment by jaeilepp** [comment]
Nope.

**Comment by jasmainak** [comment]
@jaeilepp ready for review. Should work now :)

**Comment by jaeilepp** [comment]
LGTM

**Comment by Eric89GXL** [comment]
LGTM, but I think we more or less copy these files from `sphinx-gallery`, right @agramfort? If so, then we should probably make a PR to them, get that merged, and then make the equivalent changes here.

**Comment by agramfort** [comment]
@jaeilepp merge if it works for you. You're the one fighting with the doc these days

**Comment by jasmainak** [comment]
@Eric89GXL maybe you are talking about this: https://github.com/sphinx-gallery/sphinx-gallery/pull/85? It's already merged in sphinx_gallery

**Comment by Eric89GXL** [comment]
Ahh yeah probably. I thought we also copied our `Makefile` from them, but that must not be the case.

**Comment by jaeilepp** [comment]
Thx @jasmainak

**Pull request merged**